sycl/test/check_device_code/spec_const_usid_str.cpp

```diff
@@ -0,0 +1,16 @@
+// REQUIRES: hip
```

Comment:
Why do we only test it for HIP?

Reply:
> Why do we only test it for HIP?

Added spirv64 test. Only amdgpu and spirv64's TargeInfo implements getConstantAddressSpace, so we can only add test for these two targets.